Generic aliases generate a C error

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** V 0.2.2 0e08d15.c4da74d
**OS:** Bodhi 6.0.0

**What did you do?**
```v
$ cat generic_alias.v
type Ptr<T> = &T

fn main() {
}
```

**What did you expect to see?**
a V error, since generic aliases are not allowed
 
**What did you see instead?**
```c
/tmp/v/generic_alias.10375535247574441478.tmp.c:798:9: error: unknown type name 'T'
typedef T* main__Ptr;
        ^
1 error generated.
```

--- a/vlib/v/parser/parser.v
+++ b/vlib/v/parser/parser.v
@@ -3107,6 +3107,7 @@ fn (mut p Parser) type_decl() ast.TypeDecl {
 	}
 	mut sum_variants := []ast.TypeNode{}
 	generic_types := p.parse_generic_type_list()
+	decl_pos_with_generics := decl_pos.extend(p.prev_tok.position())
 	p.check(.assign)
 	mut type_pos := p.tok.position()
 	mut comments := []ast.Comment{}
@@ -3179,6 +3180,10 @@ fn (mut p Parser) type_decl() ast.TypeDecl {
 		}
 	}
 	// type MyType = int
+	if generic_types.len > 0 {
+		p.error_with_pos('generic type aliases are not yet implemented', decl_pos_with_generics)
+		return ast.AliasTypeDecl{}
+	}
 	parent_type := first_type
 	parent_sym := p.table.get_type_symbol(parent_type)
 	pidx := parent_type.idx()

--- a/vlib/v/parser/tests/generic_type_alias_decl.out
+++ b/vlib/v/parser/tests/generic_type_alias_decl.out
@@ -1,0 +1,3 @@
+vlib/v/parser/tests/generic_type_alias_decl.vv:1:1: error: generic type aliases are not yet implemented
+    1 | type Pointer<T> = &T
+      | ~~~~~~~~~~~~~~~

--- a/vlib/v/parser/tests/generic_type_alias_decl.vv
+++ b/vlib/v/parser/tests/generic_type_alias_decl.vv
@@ -1,0 +1,1 @@
+type Pointer<T> = &T

